XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -232,11 +234,23 @@ protected override void Run()
                     throw new CancelledException();
             }
 
-            log.DebugFormat("Extracting XenServer patch '{0}'", updateName);
-            Description = string.Format(Messages.DOWNLOAD_AND_EXTRACT_ACTION_EXTRACTING_DESC, updateName);
-            ExtractFile();
-            log.DebugFormat("Extracting XenServer patch '{0}' completed", updateName);
-
+            if (nonExtractUpdate)
+            {
+                string newFilePath = Path.Combine(Path.GetDirectoryName(outputFileName), updateName);
+                if (File.Exists(newFilePath))
+                    File.Delete(newFilePath);

```

Comment:
Can we include some exception handling for these IO operations (Delete and Move).